tensorboard/webapp/feature_flag/views/feature_flag_page_test.ts

```diff
@@ -0,0 +1,232 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {CommonModule} from '@angular/common';
+import {NO_ERRORS_SCHEMA} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+import {Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+import {State} from '../../app_state';
+import {
+  allFeatureFlagOverridesReset,
+  featureFlagOverrideChanged,
+  featureFlagOverridesReset,
+} from '../actions/feature_flag_actions';
+import {
+  getDefaultFeatureFlags,
+  getOverriddenFeatureFlags,
+} from '../store/feature_flag_selectors';
+import {buildFeatureFlagState, buildState} from '../store/testing';
+import {FeatureFlags} from '../types';
+import {FeatureFlagPageModule} from './feature_flag_module';
+import {FeatureFlagPageComponent} from './feature_flag_page_component';
+import {
+  FeatureFlagPageContainer,
+  TEST_ONLY,
+} from './feature_flag_page_container';
+import {FeatureFlagOverrideStatus} from './types';
+
+describe('feature_flag_page_container', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+  let fixture: ComponentFixture<FeatureFlagPageContainer>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [FeatureFlagPageContainer],
+      imports: [CommonModule, FeatureFlagPageModule],
+      providers: [
+        provideMockStore({
+          initialState: buildState(
```

Comment:
An alternative to constructing initialState like this that we usually use is to just override all the selectors that the components rely on. I'm fine with what you have since its simple but just wanted to let you know in case you find that simpler in this case.

Reply:
I tried that initially and I was getting errors from the selectors indicating that state was undefined.